### PR TITLE
Adding SvtRawHits Baseline Fit Analysis

### PR DESCRIPTION
**Pull request by alspellm**
SvtBl2DAnaProcessor generates 2D histograms of rawsvthits.
SvtBlFitHistoProcessor generates Flat Tuple of baseline fit parameters.

### CODE REVIEW

File: analysis/include/HistoManager.h [hunk 1]
```diff
@@ -77,7 +78,7 @@ class HistoManager {
         void Fill1DHisto(const std::string& histoName, float value, float weight=1.);
         void Fill2DHisto(const std::string& histoName, float valuex, float valuey, float weight=1.);
 
-        virtual void GetHistosFromFile(TFile* inFile, const std::string& name,const std::string& folder = "");
+        virtual void GetHistosFromFile(TFile* inFile, const std::vector<std::string>& name,const std::string& folder = "");
```
Comment (cbravo135) [inline analysis/include/HistoManager.h:81]: This is potentially dangerous since it modifies an already existing method. Would be safer to just overload the method with the vector version and have both available.

File: analysis/include/ModuleMapper.h [hunk 2]
```diff
@@ -9,7 +9,8 @@
 class ModuleMapper {
 
  public:
-  ModuleMapper(const int year = 2019);
+  ModuleMapper();
+  ModuleMapper(const int year);
```
Comment (cbravo135) [inline analysis/include/ModuleMapper.h:13]: Removing a default is dangerous.

File: analysis/src/HistoManager.cxx [hunk 3]
```diff
@@ -44,67 +47,116 @@ void HistoManager::Clear() {
...
+                        hist.value().at("xtitle"),hist.value().at("binsX"),hist.value().at("minX"),hist.value().at("maxX"),
+                        hist.value().at("ytitle"),hist.value().at("binsY"),hist.value().at("minY"),hist.value().at("maxY"));
+            }
+
+
+        }//loop on config
+    }
 }
 
-void HistoManager::GetHistosFromFile(TFile* inFile, const std::string& name, const std::string& folder) {
```
Comment (cbravo135) [inline analysis/src/HistoManager.cxx:93]: ok, it looks like name wasn't really used before at all.

File: event/module.modulemap [hunk 4]
```diff
@@ -0,0 +1,6 @@
+module "event" {
```
Comment (cbravo135) [inline event/module.modulemap]: Why is this file here at all? What is this doing?

File: processors/config/anaSvtBl2D_cfg.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import HpstrConf
+import sys
+import os
+
+# Use the input file to set the output file name
+infile = sys.argv[1].strip()
```
Comment (cbravo135) [inline processors/config/anaSvtBl2D_cfg.py]: Please change this to the established way of using -i and -o to specify the input and output filenames. This is ok for your own personal local configs, but should not be like this in the repo.

File: processors/config/anaSvtRawHitsMC_cfg.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import HpstrConf
+import sys
+import os
+
+# Use the input file to set the output file name
+infile = sys.argv[1].strip()
```
Comment (cbravo135) [inline processors/config/anaSvtRawHitsMC_cfg.py:6]: Same comment as above, please use -i and -o options to pass in the input and output filenames from the command line.

File: processors/config/rawSvtHits_cfg.py [hunk 7]
```diff
@@ -4,7 +4,6 @@
 # Use the input file to set the output file name
 lcio_file = sys.argv[1].strip()
```
Comment (cbravo135) [inline processors/config/rawSvtHits_cfg.py:5]: While you are at it, please update this config file also to use -i and -o options.

File: processors/include/ClusterOnTrackAnaProcessor.h [hunk 8]
```diff
@@ -5,7 +5,7 @@
 #include "HpsEvent.h"
 #include "Track.h"
 #include "TrackerHit.h"
-
+#include "Collections.h"
```
Comment (cbravo135) [inline processors/include/ClusterOnTrackAnaProcessor.h]: Why are you adding this, we are trying to deprecate Collections.h. Try removing it and see if anything breaks without it.

File: processors/src/ClusterOnTrackAnaProcessor.cxx [hunk 9]
```diff
@@ -1,6 +1,6 @@
 #include "ClusterOnTrackAnaProcessor.h"
 #include "TBranch.h"
-
+#include "Collections.h"
```
Comment (cbravo135) [inline processors/src/ClusterOnTrackAnaProcessor.cxx]: Try removing this. We might need to pass in some config params to be able to do that, but you can see examples of how this is done in other processors.

File: pyplot/baselineFits/.gitignore [hunk 10]
```diff
@@ -0,0 +1,2 @@
+old/
```
Comment (cbravo135) [inline pyplot/baselineFits/.gitignore:1]: It is not a clean style to put an ignore here to ignore stuff only you have locally. Please remove and change workflow to not run these directly in the project source directories.

File: pyplot/baselineFits/iterativeGausFit.py [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+
+def savePNG(canvas,directory,name):
+        canvas.Draw()
+        canvas.SaveAs("%s/%s.png"%(directory,name))
+        canvas.Close()
+
+
+
+######################################################################################################
+directory = "/home/alic/src/hpstr/pyplot/baselineFits/fit_data/"
```
Comment (cbravo135) [inline pyplot/baselineFits/iterativeGausFit.py]: Try to avoid hardcoding stuff like this. You should be able to pass this in via command line.

File: pyplot/baselineFits/minwinFit.py [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+def savePNG(canvas,directory,name):
+        canvas.Draw()
+        canvas.SaveAs("%s/%s.png"%(directory,name))
+        canvas.Close()
+
+
+
+######################################################################################################
+import numpy as np
+directory = "/home/alic/src/hpstr/pyplot/baselineFits/fit_data/"
```
Comment (cbravo135) [inline pyplot/baselineFits/minwinFit.py]: Please avoid hardcoding in these scripts so it is more transparent for others to run.

**Review by cbravo135 — CHANGES_REQUESTED** [review]
I do greatly appreciate you adding all the plotting scripts to the repo. Please remove hardcoded paths from them. Also, there are a few more changes I made comments in line on for you to resolve.

File: event/module.modulemap (hunk 4 above)
Comment (alspellm) [reply event/module.modulemap]: I actually have no idea how this file was created
